### DISCUSSION TITLE
Can I use the Tauri backend as an alternative to Redux or useState()?

### DISCUSSION THREAD

**Discussion by skorotkiewicz** [tauri-apps/tauri#4940] [Q&A] [answered]
Just wondering, do I need to use Redux to have a global State on React since I have a backend in Rust?

Can I use Events to transfer data between client and server and have the state work as dynamically as in Redux or useState()?
Will the changes be visible only after the component is refreshed?

Or is there any other method to use Tauri backend as "Redux" state manager?

**Comment by FabianLars** [comment, marked as answer]
You can't really replace useState(), but Redux may be possible if you currently aren't "abusing" Redux. Just keep in mind that setting/receiving the state will be slightly slower because of the IPC round trip. Shouldn't be noticeable unless you're sending >1mb payloads.
Oh and not to forget, Tauri's api are async, mentioning it in case it's a no-go for you.

That said, i do use tauri::State as a cache for my routes too, especially to handle page refreshes.
